File: test/leetcode/q11_container_with_most_water/SolutionTest.java
package leetcode.q11_container_with_most_water;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {

    @Test
    void maxArea_1() {
        validate(new int[]{1, 8, 6, 2, 5, 4, 8, 3, 7}, 49);
    }

    @Test
    void maxArea_2() {
        validate(new int[]{1, 1}, 1);
    }

    private void validate(int[] height, int expected) {
        assertEquals(expected, new Solution().maxArea(height));
    }
}

File: test/leetcode/q7_Reverse_Integer/SolutionTest.kt
package leetcode.q7_Reverse_Integer

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class SolutionTest {

    @Test
    fun reverse_1() = validate(123, 321)

    @Test
    fun reverse_2() = validate(-123, -321)

    @Test
    fun reverse_3() = validate(120, 21)

    @Test
    fun reverse_4() = validate(0, 0)

    private fun validate(x: Int, expected: Int) {
        assertEquals(expected, Solution().reverse(x))
    }
}